src/controller/proxy/controller.go

```diff
@@ -161,26 +164,38 @@ func (c *controller) UseLocalManifest(ctx context.Context, art lib.ArtifactInfo,
 	}
 
 	var content []byte
-	if c.cache != nil {
-		err = c.cache.Fetch(getManifestListKey(art.Repository, string(desc.Digest)), &content)
-		if err == nil {
-			log.Debugf("Get the manifest list with key=cache:%v", getManifestListKey(art.Repository, string(desc.Digest)))
-			return true, &ManifestList{content, string(desc.Digest), manifestlist.MediaTypeManifestList}, nil
-		}
-		if err == cache.ErrNotFound {
-			log.Debugf("Digest is not found in manifest list cache, key=cache:%v", getManifestListKey(art.Repository, string(desc.Digest)))
-		} else {
-			log.Errorf("Failed to get manifest list from cache, error: %v", err)
-		}
+	var contentType string
+	if c.cache == nil {
+		return a != nil && string(desc.Digest) == a.Digest, nil, nil // digest matches
+	}
+
+	err = c.cache.Fetch(manifestListKey(art.Repository, string(desc.Digest)), &content)
+	if err == cache.ErrNotFound {
```

Comment:
In the original logic, when `err == cache.ErrNotFound`, no error is returned, but now it returns an error because it also matches `if err != nil` code block

Reply:
updated